File: app/src/main/java/com/falcon/restaurants/screens/common/ScreensNavigator.java
package com.falcon.restaurants.screens.common;

import androidx.appcompat.app.AppCompatActivity;
import com.falcon.restaurants.common.di.activity.ActivityScope;
import com.falcon.restaurants.screens.restaurant.RestaurantsActivity;
import com.falcon.restaurants.screens.meal.MealsActivity;
import com.falcon.restaurants.screens.mealdetail.MealDetailsActivity;

import javax.inject.Inject;

@ActivityScope
public class ScreensNavigator {

    private AppCompatActivity fromActivity;

    @Inject
    public ScreensNavigator(AppCompatActivity fromActivity) {
        this.fromActivity = fromActivity;
    }

    public void toCategoriesActivity(String restaurantId){
        RestaurantsActivity.start(fromActivity, restaurantId);
    }

    public void toMealsActivity(String restaurantId) {
        MealsActivity.start(fromActivity, restaurantId);
    }

    public void toMealDetailsActivity(String mealId){
        MealDetailsActivity.start(fromActivity, mealId);
    }
}

File: app/src/main/java/com/falcon/restaurants/screens/common/ScreensNavigator.kt
package com.falcon.restaurants.screens.common;

import androidx.appcompat.app.AppCompatActivity;
import com.falcon.restaurants.common.di.activity.ActivityScope;
import com.falcon.restaurants.screens.restaurant.RestaurantsActivity;
import com.falcon.restaurants.screens.meal.MealsActivity;
import com.falcon.restaurants.screens.mealdetail.MealDetailsActivity;

import javax.inject.Inject;

@ActivityScope
class ScreensNavigator @Inject constructor( val fromActivity: AppCompatActivity){

    fun toCategoriesActivity(restaurantId: String){
        RestaurantsActivity.start(fromActivity, restaurantId)
    }

    fun toMealsActivity(restaurantId: String) {
        MealsActivity.start(fromActivity, restaurantId)
    }

    fun toMealDetailsActivity(mealId: String){
        MealDetailsActivity.start(fromActivity, mealId)
    }
}
